<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="mainWidget">
   <layout class="QHBoxLayout" name="mainHorizontalLayout">
    <item>
     <spacer name="mainLeftSpacer">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QVBoxLayout" name="mainVerticalLayout">
      <item>
       <layout class="QHBoxLayout" name="functionHorizontalLayout">
        <item>
         <spacer name="functionLeftSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="functionVerticalLayout">
          <item>
           <widget class="QLabel" name="functionLabel">
            <property name="text">
             <string>Функція:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="function1RadioButton">
            <property name="text">
             <string>3sin(5x) - 2cos(3x) + 3</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="function2RadioButton">
            <property name="text">
             <string>5^(34+x^5) - 5^(1-x^2)</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="functionRightSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="pointsParametersLayout">
        <item>
         <widget class="QLabel" name="startLabel">
          <property name="text">
           <string>Початок</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="startSpinBox">
          <property name="decimals">
           <number>10</number>
          </property>
          <property name="minimum">
           <double>-1000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="endLabel">
          <property name="text">
           <string>Кінець</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="endSpinBox">
          <property name="decimals">
           <number>10</number>
          </property>
          <property name="minimum">
           <double>-1000.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="pointCountLabel">
          <property name="text">
           <string>Кількість точок</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="pointCountSpinBox">
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item alignment="Qt::AlignmentFlag::AlignHCenter">
       <widget class="QPushButton" name="calculationButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Обчислити</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="plotLayout">
        <item>
         <widget class="Plot" name="plot" native="true">
          <property name="minimumSize">
           <size>
            <width>300</width>
            <height>300</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">border: 2px solid black;</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="plotParametersLayout">
          <item>
           <spacer name="plotParametersVerticalSpacer1">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="lineWidthLabel">
            <property name="text">
             <string>Товщина лінії:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="lineWidthSpinBox">
            <property name="singleStep">
             <double>0.250000000000000</double>
            </property>
            <property name="value">
             <double>2.000000000000000</double>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="plotParametersVerticalSpacer2">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="lineColorLabel">
            <property name="text">
             <string>Колір:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="lineRedLayout">
            <item>
             <widget class="QLabel" name="lineRedLabel">
              <property name="text">
               <string>R</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="lineRedSlider">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="lineGreenLayout">
            <item>
             <widget class="QLabel" name="lineGreenLabel">
              <property name="text">
               <string>G</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="lineGreenSlider">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="lineBlueLayout">
            <item>
             <widget class="QLabel" name="lineBlueLabel">
              <property name="text">
               <string>B</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="lineBlueSlider">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="lineAlphaLayout">
            <item>
             <widget class="QLabel" name="lineAlphaLabel">
              <property name="text">
               <string>A</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="lineAlphaSlider">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>255</number>
              </property>
              <property name="orientation">
               <enum>Qt::Orientation::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="plotParametersVerticalSpacer3">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="lineStyleLabel">
            <property name="text">
             <string>Стиль лінії:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="lineStyleSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>5</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
            <property name="sliderPosition">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="tickInterval">
             <number>0</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="plotParametersVerticalSpacer4">
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="resultTextEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="mainRightSpacer">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Plot</class>
   <extends>QWidget</extends>
   <header>Plot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>calculationButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>releaseCalculationButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>394</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineRedSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changePlotStyle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>73</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineWidthSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changePlotStyle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>642</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>54</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineGreenSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changePlotStyle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>40</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineBlueSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changePlotStyle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>36</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineStyleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changePlotStyle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>507</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>44</x>
     <y>475</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineAlphaSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>changePlotStyle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>554</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>60</x>
     <y>545</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>releaseCalculationButton()</slot>
  <slot>changePlotStyle()</slot>
 </slots>
</ui>
